e2e/testcases/lifecycle_directives_test.go

```diff
@@ -61,23 +60,22 @@ func TestPreventDeletionNamespace(t *testing.T) {
 	nt.Must(rootSyncGitRepo.CommitAndPush("declare Namespace with prevent deletion lifecycle annotation"))
 	nt.Must(nt.WatchForAllSyncs())
 
-	err = nt.Validate("shipping", "", &corev1.Namespace{},
+	rsObj := &v1beta1.RootSync{}
+	nt.Must(nt.KubeClient.Get(rootSyncNN.Name, rootSyncNN.Namespace, rsObj))
+	applySetID := core.GetLabel(rsObj, metadata.ApplySetParentIDLabel)
+
+	nt.Must(nt.Validate("shipping", "", &corev1.Namespace{},
 		testpredicates.NotPendingDeletion,
 		testpredicates.HasAnnotation(common.LifecycleDeleteAnnotation, common.PreventDeletion),
-		testpredicates.HasAllNomosMetadata())
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+		testpredicates.HasAllNomosMetadata(),
```

Comment:
Perhaps `HasAllNomosMetadata` should include a check for the apply set label, and `NoConfigSyncMetadata` should include a check that the label is missing.

Is there a case where an object should have other CS metadata but not the apply set label?

Reply:
The ApplySet label isn't technically "Config Sync metadata", and I didn't want to have to pass the ApplySetID to the Config Sync metadata methods, so I left it out and gave it its own method in the impl and the tests.

One critical difference is that HasConfigSyncMetadata does not detect the ApplySet label, partly because in order to do so it would need to check the label value, not just the key. So the ApplySet label is only removed if other CS metadata is present AND the ApplySet ID matches.

Reply:
I considered adding a HasApplySetPartOfLabel predicate, but it seemed pretty superfluous given that the HasLabel predicate is the same number of code lines.